src/fiatconnect/BankDetailsScreen.tsx

```diff
@@ -0,0 +1,275 @@
+const SCHEMA_TO_FIELD_METADATA_MAP = {
+  AccountNumber: [
+    {
+      name: 'accountName',
+      label: i18n.t('fiatAccountSchema.accountName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
+    },
+    {
+      name: 'institutionName',
+      label: i18n.t('fiatAccountSchema.institutionName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^\d{10}$/,
+      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
+    },
+  ],
+}
+
+const BankDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set())
+  const inputRefs = useRef<string[]>([textValue])
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('bankDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: { [k: string]: string } = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      await addNewFiatAccount(providerURL, fiatAccountSchema, body)
+        .then((data) => {
+          // TODO Tracking here
+          dispatch(showMessage(t('bankDetailsScreen.addFiatAccountSuccess')))
+        })
+        .catch((error) => {
+          // TODO Tracking here
+          if (error === FiatConnectError.ResourceExists) {
+            dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
+          }
+          Logger.error(TAG, `Error adding fiat account: ${errors}`)
+        })
+
+      // TODO: navigate to the next screen
+    }
+  }
+
+  const onPressSelectPaymentOption = () => {
+    // TODO: tracking here
+    // todo: naviagte to select payment options screen
+  }
+
+  const validateInput = () => {
+    setValidInputs(false)
```

Comment:
Does this cause some weird flashing of the 'Next' button as the user types their fields in? Maybe it should only be set to invalid if an error is actually found

Reply:
to be honest, the button is covered by input keyboard when user is typing.

If it's not covered, I imagine it would only flash if the field value can change between invalid and valid as user type? which feels unlikely to me?